### ISSUE TITLE
Stack overflow outputting "Segmentation fault."

### ISSUE THREAD

**Issue by Erlite** [dotnet/coreclr#12315]
So basically I had an attribute, which contained this string with a get/set:

```cs
private string Message
{
    get
    {
        return this.Message;
    }
    set
    {
        this.Message = value;
    }
}
```

Upon compiling, any attempt to run it using `dotnet run` would just silently fail. However, trying to execute the dll with `dotnet exec` would output **Segmentation fault.** to the console. Figured out that string was causing a stack overflow. Shouldn't it throw an exception instead of a segfault?

**Comment by danmosemsft** [comment]
@janvorli

**Comment by janvorli** [comment]
@Erlite have you tried that with dotnet core 2.0? We were not reporting most of the cases of stack overflow before that version since the OS just kills the process in such case without our runtime getting notified.
In 2.0, I have added mechanism to allow us to print message reporting stack overflow. I have tried your example with 2.0 and I got this:
```
../dotnet bin/Debug/netcoreapp2.0/testso.dll
Process is terminating due to StackOverflowException.
Aborted (core dumped)
```
